### PR TITLE
Dev azurestackhci microsoft.azure stack hci 2022 10 01

### PR DESCRIPTION
**Pull request by aditya-rs**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:43]: >v2 [](http://example.com/codeflow?start=69&length=2)

Recommend using v3 common types for new APIs. This comment applies throughout this file and other added code of the PR.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:219]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

Use string enum for this. Must include terminal states (Succeeded, Failed, Canceled). Transition states are up to you.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:224]: >publisherId [](http://example.com/codeflow?start=9&length=11)

It helps to describe what kind of Id this is. If it's an ARM resource, then "publisherResourceId" would be recommended (and updated in the description). If it's a different kind of Id, then adding some info in the description would be helpful.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:250]: >catalogPlanId [](http://example.com/codeflow?start=9&length=13)

Suggest catalogPlanResourceId and/or update description.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:254]: >marketplaceSkuId [](http://example.com/codeflow?start=9&length=16)

Same.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/publishers.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/publishers.json:43]: >v2 [](http://example.com/codeflow?start=69&length=2)

Recommend v3 common types

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/publishers.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/publishers.json:162]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

String enum

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:180]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

Model as string enum per earlier comment.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:185]: >publisherId [](http://example.com/codeflow?start=9&length=11)

Same as earlier comments, include resourceId in name/description or info about whatever type of Id it is.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:189]: >offerId [](http://example.com/codeflow?start=9&length=7)

Same here and for catalogPlanId, marketplaceSkuId.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json:165]: >Describes Machine Extension Properties [](http://example.com/codeflow?start=26&length=38)

Seems wrong

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json:43]: >v2 [](http://example.com/codeflow?start=69&length=2)

Recommend v3 common types

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json:191]: >timeStarted [](http://example.com/codeflow?start=9&length=11)

This and lastUpdatedTime look like they belongs in systemData of the resource. The systemData property is already part of the ProxyResource definition in v3 common types. Ref: [https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html](https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html).

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateRuns.json:197]: >lastUpdatedTime [](http://example.com/codeflow?start=9&length=15)

Same

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:43]: >v2 [](http://example.com/codeflow?start=69&length=2)

v3

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:58]: >responses [](http://example.com/codeflow?start=9&length=9)

Also need final 200 response.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:119]: >updates/{updateName} [](http://example.com/codeflow?start=129&length=20)

How are these created? I find it slightly odd that they aren't created here, but there is a POST API here to "apply" them. I'm assuming all of the new APIs in this PR are passing through from an external source, but it seems strange that you need to talk to that external source to create an update, then move to this API to "apply" it. Can you explain this a bit?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:98]: x-ms-pageable is recommended, even if you haven't yet implemented the paging. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html?q=x-ms-pageable#oapi009-always-add-x-ms-pageable-to-list-calls](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html?q=x-ms-pageable#oapi009-always-add-x-ms-pageable-to-list-calls)

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:242]: >installedDate [](http://example.com/codeflow?start=9&length=13)

Is there a difference between "install" and "apply"?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:334]: Does this have a fixed set of values? If so, it should be modeled as a string enum.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:37]: >default [](http://example.com/codeflow?start=145&length=7)

Current guidance is to model this as an enum parameter with one value, rather than hard-coded path segment.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:36]: >paths [](http://example.com/codeflow?start=3&length=5)

Still need a collection GET API for updateSummaries (even though you only have one).

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:153]: >lastUpdated [](http://example.com/codeflow?start=9&length=11)

Belongs in systemData.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:133]: >lastChecked [](http://example.com/codeflow?start=9&length=11)

Belongs in systemData.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:242]: No

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:334]: Values are not fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:37]: For current resources in our RP, we use default in the path (as a non-enum) and would prefer to keep it that way if possible

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:184]: Not a fixed set of constant strings

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:125]: These are created internally. The intention is to not let clients (customers) make PUT calls to these resources. They can be viewed with GET, and actions can be taken with POST

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:224]: Description already added

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:250]: Will not fix

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:254]: Will not fix

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:185]: Description already present

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (aditya-rs) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:189]: Will not fix

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (bsomeshekar) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:219]: Since these are browse api's and provisioningState object is kind of mandate for all resource types so we added this obj as string type with default value set to 'succeeded'. Do you want us to change to enum and define only one value(which is 'succeeded') ?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (bsomeshekar) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:180]: Same comment as above

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (mentat9) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:55]: >responses [](http://example.com/codeflow?start=9&length=9)

Need x-ms-pageable

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:219]: If your backing resource doesn't communicate any provisioning states, I don't think you should include the property here. Since your APIs don't support PUT/PATCH/DELETE, it's only meaningful if you can translate similar state semantics from your backing resource (e.g., what value do you provide here if someone GETs the resource while it is in the middle of being PUT or PATCHed?). If you decide you want to keep it, you need to define it as string enum with at least the 3 terminal states: Succeeded, Failed, and Canceled. You are welcome to add additional transition states as needed.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:224]: But it's ambiguous. If it's not an ARM resource Id, then what is it? A guid? A user-selected name, something else? It's helpful for users of your APIs to have more than "publisher Id" to go on.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:250]: Recommend a better description. It gives no additional information not already in the name. This comment applies to all of the *Id properties I previously identified. The word Id is inherently ambiguous. We use resourceId to disambiguate it for ARM resources. If these are some other kind of resource, it is very helpful to disambiguate the way they are identified in the description. If they are ARM resources, but you don't want to use resource in the name, at least call it a resource Id in the description (or Azure resource Id, not ARM).

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:180]: Same as above. Probably best to remove it, but it still needs to be a string enum if it remains.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:101]: Why did you add and then remove x-ms-pageable? You added the nextLink property: this looks like a mistake. You need x-ms-pageable to get the paging semantics.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:250]: Got it. Then this property should go in the systemData.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (agrbhuvnesh) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:219]: Actually, this spec has been already part of 
https://github.com/Azure/azure-rest-api-specs-pr/blob/RPSaaSMaster/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-04-01-preview/offers.json#L219

and rolled out for public preview, shall it be acceptable to modify now 
will that not be considered as breaking change?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json
Comment (agrbhuvnesh) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updateSummaries.json:55]: updated.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:219]: This is why we ask you to include the current/latest API version files as your first commit. Reviews can't tell what to review if you don't do that. This appeared to be a newly added file/API in this PR, so I reviewed it as such.

I'm OK with leaving this as string for this API version, given that it's already published as unrestricted string. In your next API version, I think you should be able to change it to string enum (requires breaking change review) without impacting customers.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/publishers.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/publishers.json:162]: Resolving since this API already shipped as string.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/skus.json:180]: Resolving, since this API already shipped. Recommend fixing it in the next API version.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/updates.json:101]: Resolving this to unblock manifest PR. IMPORTANT: be sure you add x-ms-pageable before merging and generating SDKs.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (mentat9) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:254]: Please improve the description by disambiguating the Id term (e.g. Resource Id, Guid, Client selected name, etc.) before merging this PR.

**Comment by mentat9** [comment]
Signing off with comments to unblock. Please address remaining comments before merging this PR These two are particularly important:
https://github.com/Azure/azure-rest-api-specs/pull/20924#discussion_r984841941
https://github.com/Azure/azure-rest-api-specs/pull/20924#discussion_r984842088

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json
Comment (bsomeshekar) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json:224]: PublisherId is one of the attributes to determine marketplace image offer. A marketplace image in azure has publisher, offer, sku and version attributes which uniquely identify a given image offer.
We tried to align with existing compute RP marketplace APIs, so we followed similar naming convention for publishers, offers and skus resource types in our RP as well.

**Review by jordan13game — APPROVED** [review]

**Pull request merged**
